### ISSUE TITLE
Rule `require-default-props` with Flow type definition assignment

### ISSUE THREAD

**Issue by wyze** [yannickcr/eslint-plugin-react#1017]
Using v6.9.0 with code like the following:

```js
// @flow

import type { Message } from '../types'

type Props = Message

const Component = ({ message }: Props) => (
  <div>{message}</div>
)

export default Component
```

Crashes with:

```
Cannot read property 'filter' of undefined
TypeError: Cannot read property 'filter' of undefined
    at getPropTypesFromTypeAnnotation ($PWD/node_modules/eslint-plugin-react/lib/rules/require-default-props.js:178:29)
    at Object.handleStatelessComponent ($PWD/node_modules/eslint-plugin-react/lib/rules/require-default-props.js:288:42)
    at EventEmitter.updatedRuleInstructions.(anonymous function) ($PWD/node_modules/eslint-plugin-react/lib/util/Components.js:581:75)
    at emitOne (events.js:101:20)
    at EventEmitter.emit (events.js:188:7)
    at NodeEventGenerator.enterNode ($PWD/node_modules/eslint/lib/util/node-event-generator.js:39:22)
    at CodePathAnalyzer.enterNode ($PWD/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:607:23)
    at CommentEventGenerator.enterNode ($PWD/node_modules/eslint/lib/util/comment-event-generator.js:97:23)
    at Controller.enter ($PWD/node_modules/eslint/lib/eslint.js:928:36)
    at Controller.__execute ($PWD/node_modules/estraverse/estraverse.js:397:31)
```

I have a fix for this, and will submit the PR shortly.

**Comment by wyze** [comment]
Also, in debugging this, it looks like it is not specific to importing the type. It will also throw the same error if you do something like this:

```js
// @flow

type Message = {
  message: string,
}
type Props = Message

const Component = ({ message }: Props) => (
  <div>{message}</div>
)

export default Component
```

**Comment by wyze** [comment]
Closing as the PR to fix was merged and released in 6.10.0.
